Objects/longobject.c

```diff
@@ -2732,6 +2733,7 @@ long_rem(PyLongObject *a, PyLongObject *b, PyLongObject **prem)
     else {
         /* Slow path using divrem. */
         Py_XDECREF(x_divrem(a, b, prem));
+        *prem = maybe_small_long(*prem);
```

Comment:
It surprised me initially that it was okay to call `maybe_small_long` on a value that might be `NULL`, but I see that other parts of `Objects/longobject.c` also do that (and it makes sense from a convenience perspective).